### PR TITLE
[ELF,test] Add test for R_AARCH64_* implicit addends

### PR DESCRIPTION
**Pull request by MaskRay**
to support certain static relocations in the REL format. See #87328 for
the armasm need.

Note: `R_AARCH64_{ABS64,PREL32,PREL64}` have been implemented by https://reviews.llvm.org/D120535

### CODE REVIEW

**Comment by artagnon** [comment]
LGTM, but I have a question: will `not` suffice, or do we require `not --crash`? Is `REQUIRES: asserts` also needed?

File: lld/test/ELF/aarch64-reloc-implicit-addend.test [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    Flags:   [ SHF_ALLOC ]
+    AddressAlign: 4
+    Content: 00ffffff00ffffff00ffffff00ffffff00ffffff00ffffff
+  - Name:    .rel.prel
+    Type:    SHT_REL
+    Link:    .symtab
+    Info:    .prel
+    Relocations:
+      - {Offset:  0, Symbol: .prel, Type: R_AARCH64_PREL64}
+      - {Offset:  4, Symbol: .prel, Type: R_AARCH64_PREL32}
```
Comment (artagnon) [inline lld/test/ELF/aarch64-reloc-implicit-addend.test:57]: Please update `relocation-rel-format.test` like in the original patch, so these aren't checked twice.

File: lld/test/ELF/aarch64-reloc-implicit-addend.test [hunk 2]
```diff
@@ -0,0 +1,86 @@
+## Test certain REL relocation types generated by legacy armasm.
```
[the remark this answers is not in the document]
Comment (MaskRay) [reply lld/test/ELF/aarch64-reloc-implicit-addend.test:1]: We don't need it. yaml2obj and ld.lld (without LTO) do not need llvm/lib/Target/ code.

File: lld/test/ELF/aarch64-reloc-implicit-addend.test (hunk 1 above)
Comment (MaskRay) [reply lld/test/ELF/aarch64-reloc-implicit-addend.test:57]: I think it is fine to keep `relocation-rel-format.test` unchanged. That file shows that we support REL for certain targets.

**Review by artagnon — APPROVED** [review]

**Comment by MaskRay** [comment]
> LGTM, but I have a question: will `not` suffice, or do we require `not --crash`? Is `REQUIRES: asserts` also needed?

getImplicitAddend is used by both static relocations and dynamic relocations. The dynamic relocations are only for `config->checkDynamicRelocs` checks, where there is a `#ifdef NDEBUG` difference.

Static relocations don't need `REQUIRES: asserts`.

**Pull request merged**